dds/DCPS/DataReaderImpl.cpp

```diff
@@ -856,6 +856,12 @@ DDS::ReturnCode_t DataReaderImpl::set_qos(const DDS::DataReaderQos& qos_arg)
     // Make sure Data Representation QoS is initialized
     topic_servant_->check_data_representation_qos(qos_.representation.value);
 
+    if (qos.representation.value.length() == 0) {
+      qos.representation.value.length(2);
+      qos.representation.value[0] = DDS::XCDR_DATA_REPRESENTATION;
+      qos.representation.value[1] = UNALIGNED_CDR_DATA_REPRESENTATION;
+    }
```

Comment:
Actually I think these particular lines of code should be removed. The line of code before these lines should be defaulting the QoS to compare to, but it has a typo.

Instead of

```
topic_servant_->check_data_representation_qos(qos_.representation.value);
```

it should be

```
topic_servant_->check_data_representation_qos(qos.representation.value);
```

The equivalent lines in `DataWriterImpl::set_qos` are correct.